### PR TITLE
CORE-8910 Merge the Collaborators and Collaborator Lists into one view

### PR DESCRIPTION
**Pull request by ash3rz**
Majority of the changes are 
* Moving items from GroupView to ManageCollaboratorsView
* Deleting the old GroupView code
* Merging the GroupView appearance into the ManageCollaboratorsAppearance
* Setting up a parent callback so that when a user hits the Delete button, it will do all the logic to delete any users selected (one call no matter how many users selected) and delete any groups selected (the number of calls is equivalent to the number of selected groups).

### CODE REVIEW

File: de-lib/src/main/java/org/iplantc/de/collaborators/client/presenter/ManageCollaboratorsPresenter.java [hunk 1]
```diff
@@ -372,25 +484,29 @@ public void onGroupSaved(GroupSaved event) {
...
             }
         });
     }
 
+    ParentDeleteSubjectsCallback getParentDeleteSubjectsCallback() {
+        return new ParentDeleteSubjectsCallback();
+    }
+
     AddMemberToGroupCallback getAddMemberToGroupCallback() {
         return new AddMemberToGroupCallback();
```
Comment (ash3rz) [inline de-lib/src/main/java/org/iplantc/de/collaborators/client/presenter/ManageCollaboratorsPresenter.java]: Sure, I can change the method names

File: de-lib/src/main/java/org/iplantc/de/collaborators/client/presenter/callbacks/ParentDeleteSubjectsAbstractCallback.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    int doneCount;
+    List<UpdateMemberResult> successUsers = Lists.newArrayList();
+    List<Group> successGroups = Lists.newArrayList();
+    List<Throwable> failures = Lists.newArrayList();
+
+    public void setCallbackCounter(int doneCount) {
+        this.doneCount = doneCount;
+    }
+
+    public synchronized void updateCounter() {
```
Comment (ash3rz) [inline de-lib/src/main/java/org/iplantc/de/collaborators/client/presenter/callbacks/ParentDeleteSubjectsAbstractCallback.java]: Oh whoops, guess that's not necessary then 😄

File: de-lib/src/main/java/org/iplantc/de/commons/client/ErrorHandler.java [hunk 3]
```diff
@@ -107,22 +120,23 @@ public static void post(String errorSummary, Throwable caught) {
...
-        if (errorSummary == null) {
-            errorSummary = SafeHtmlUtils.fromString(appearance.error());
-        }
+        if (caughtList != null && !caughtList.isEmpty()) {
+            caughtList.forEach(caught -> {
+                GWT.log(errorSummary.asString(), caught);
 
-        if (caught != null) {
-            GWT.log(errorSummary.asString(), caught);
+                errorDetails.append(parseExceptionJson(caught) + NEWLINE);
```
Comment (psarando) [inline de-lib/src/main/java/org/iplantc/de/commons/client/ErrorHandler.java]: 👍

**Comment by ash3rz** [comment]
Thanks for the reviews!  I was thinking before I merge this, I might want to change the icon for Collaborator Lists from group to something more "list" related.  That way that group icon (or maybe even something else) could be used for Teams.

**Pull request merged**